### PR TITLE
Add python-3.10 CI

### PR DESCRIPTION
**Pull request by malfet**
First step towards https://github.com/pytorch/pytorch/issues/66424

### CODE REVIEW

File: .github/scripts/generate_ci_workflows.py [hunk 1]
```diff
@@ -503,8 +503,8 @@ def generate_workflow_file(self, workflow_template: jinja2.Template) -> None:
     ),
     CIWorkflow(
         arch="linux",
-        build_environment="linux-xenial-cuda11.3-py3.7-gcc7",
-        docker_image_base=f"{DOCKER_REGISTRY}/pytorch/pytorch-linux-xenial-cuda11.3-cudnn8-py3-gcc7",
+        build_environment="linux-xenial-cuda11.3-py3.10-gcc7",
```
Comment (atalman) [inline .github/scripts/generate_ci_workflows.py:506]: Why are we changing py3.7 workflow instead of adding new 3.10 workflow ?

File: .circleci/docker/common/install_conda.sh [hunk 2]
```diff
@@ -13,12 +13,7 @@ if [ -n "$ANACONDA_PYTHON_VERSION" ]; then
       CONDA_FILE="Miniconda2-latest-Linux-x86_64.sh"
     ;;
     3)
-      if [ "$ANACONDA_PYTHON_VERSION" = "3.6" ]; then
-        # Latest release of Conda that still supports python-3.6
-        CONDA_FILE="Miniconda3-py37_4.10.3-Linux-x86_64.sh"
-      else
-        CONDA_FILE="Miniconda3-latest-Linux-x86_64.sh"
-      fi
+      CONDA_FILE="Miniconda3-latest-Linux-x86_64.sh"
```
Comment (atalman) [inline .circleci/docker/common/install_conda.sh:16]: Thank you, for getting rid of this case

File: .github/templates/macos_binaries_workflow.yml.j2 [hunk 3]
```diff
@@ -0,0 +1,68 @@
+{% import 'common.yml.j2' as common %}
```
Comment (atalman) [inline .github/templates/macos_binaries_workflow.yml.j2:1]: This file is not really related to python 3.10. Sould it be included in this PR ?

File: .github/scripts/generate_ci_workflows.py (hunk 1 above)
Comment (malfet) [reply .github/scripts/generate_ci_workflows.py:506]: Because there are other python-3.7 cuda enabled configs, aren't they?

File: .github/templates/macos_binaries_workflow.yml.j2 (hunk 3 above)
Comment (malfet) [reply .github/templates/macos_binaries_workflow.yml.j2:1]: Good point, let me update

File: .github/scripts/generate_ci_workflows.py (hunk 1 above)
Comment (atalman) [reply .github/scripts/generate_ci_workflows.py:506]: Yes true we do have multiple 3.7 configs

**Review by atalman — APPROVED** [review]
LGTM
